bokehjs/src/coffee/models/annotations/arrow.coffee

```diff
@@ -15,13 +15,14 @@ export class ArrowView extends AnnotationView
   connect_signals: () ->
     super()
     @connect(@model.change, () => @plot_view.request_render())
-    @connect(@model.source.change, () ->
-      @set_data(@model.source)
-      @plot_view.request_render())
+    @connect(@model.source.streaming, () -> @set_data(@model.source))
+    @connect(@model.source.patching, () -> @set_data(@model.source))
+    @connect(@model.source.change, () -> @set_data(@model.source))
 
   set_data: (source) ->
     super(source)
     @visuals.warm_cache(source)
+    @plot_view.request_render()
```

Comment:
`request_render` actually precipitates a call to `throttled_render` so I'm not overly worried about it. I guess I'd rather have the mental model that  setting new data always implies a corresponding request to render in any case. `set_data` doesn't actually trigger any changes in the data source, so that can't be what was causing a render previously anyway. My guess is that lots of things are requesting renders around startup. But repeated calls to `request_render` within the throttle window should have no effect. (That was really the intent of `request_render`, to reduce the bookkeeping burden on developers).

I think if we wanted to prevent multiple renders on startup the cleanest way would just be to have a synchronization condition. I.e. *no* render ever happens until some condition is met, such as "all models have passed initialization" then after condition is met, the  flag blocking `render` is flipped, and a `request_render` is issued.